### DISCUSSION TITLE
How to write a plugin to set cookies in header of the response.

### DISCUSSION THREAD

**Discussion by cxd9974** [dotansimha/envelop#1565] [Q&A] [answered]
Hello, 
I wonder what would be the approach to write an envelop plugin that set cookies in header of the response?

I saw a related question here: https://github.com/n1ru4l/envelop/discussions/1006
But following it doesn't provide me cookie in the header I had hoped.

More context of what I'm trying to do:
https://community.redwoodjs.com/t/set-a-custom-cookie-with-a-query-or-mutation-response/4183

Thank you.

**Comment by n1ru4l** [comment, marked as answer]
Hey @cxd9974, setting a cookie is implementation specific and depends on your specific use case.

E.g. if you use Node.js you can make the request and response object part of the GraphQL context. There you can set the `cookie` header on the `http` response object (https://nodejs.org/api/http.html#responsesetheadername-value)
